server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -869,6 +871,23 @@ public Profilers getProfilers() {
         return profilers;
     }
 
+    /**
+     * Returns concurrent segment search status for the search context
+     */
+    @Override
+    public boolean isConcurrentSegmentSearchEnabled() {
+        if (clusterService != null) {
+            return indexService.getIndexSettings()
+                .getSettings()
+                .getAsBoolean(
+                    "index.search.concurrent_segment_search.enabled",
+                    clusterService.getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
```

Comment:
We need to check for FeatureFlag is enabled for concurrent search here, otherwise currently it will always return `true`

Reply:
@jed326 there is one minor tweak to here:  add `&& searcher().getExecutor() != null;` to the condition. The `ConcurrentQueryPhaseSearcher` has another check in shape of `allowConcurrentSegmentSearch()` - it has to be removed and always follow the concurrent path.